tests/emukit/core/test_loop_steps.py

```diff
@@ -22,6 +25,50 @@ def test_fixed_iteration_stopping_condition():
     assert(stopping_condition.should_stop(loop_state_mock) is True)
 
 
+def test_convergence_stopping_condition():
+    stopping_condition = ConvergenceStoppingCondition(0.1)
+
+    # check if we stop before criterion can be calculated
+    loop_state_mock = mock.create_autospec(LoopState)
+    loop_state_mock.iteration = 1
+    loop_state_mock.X = np.array([[0]])
+    assert(stopping_condition.should_stop(loop_state_mock) is False)
+
+    # check if we stop when we should not
+    loop_state_mock = mock.create_autospec(LoopState)
+    loop_state_mock.iteration = 5
+    loop_state_mock.X = np.array([[0], [10], [20], [30], [40]])
+    assert(stopping_condition.should_stop(loop_state_mock) is False)
+
+    # check if we stop when we should
+    loop_state_mock = mock.create_autospec(LoopState)
+    loop_state_mock.iteration = 5
+    loop_state_mock.X.return_value(np.array([[0], [1], [2], [3], [3.01]]))
+    assert(stopping_condition.should_stop(loop_state_mock) is True)
+
+
+def test_list_of_stopping_conditions():
```

Comment:
for that test what should ideally be mocked is stopping condition itself. that is, you would have two mock conditions in a list, and manipulate their "should_stop" outputs to test various combinations (none fires, one fires, both fire)

Reply:
btw, here is how you can make `should_stop` return different outputs in sequence: https://stackoverflow.com/questions/24897145/python-mock-multiple-return-values